app/models/article.rb

```diff
@@ -2,4 +2,7 @@ class Article < ApplicationRecord
   belongs_to :user
   has_many :bookings
   has_many :reviews
+  validates :title, presence: true, length: { in: 3..150 }
+  validates :summary, presence: true, length: { in: 50..400 }
+  validates :url, presence: true, allow_blank: false
```

Comment:
I found a regex and rails didn't like it. Will try to find a simpler one.
Agree on price, will add